middlewares/passport.middleware.js

```diff
@@ -1,46 +1,70 @@
-require("dotenv").config();
-const passport = require("passport");
-const LocalStrategy = require("passport-local");
-const users = require("../dao/user.dao");
-const { validatePassword } = require("../utilities/password.util");
+require('dotenv').config();
+const passport = require('passport');
+const LocalStrategy = require('passport-local');
+const GitHubStrategy = require('passport-github2').Strategy;
+const users = require('../dao/user.dao');
+const { validatePassword } = require('../utilities/password.util');
 
 passport.use(
-  new LocalStrategy(
-    { usernameField: "email" },
-    async (username, password, cb) => {
+  new LocalStrategy({ usernameField: 'email' }, async (username, password, cb) => {
+    try {
+      const user = await users.attemptLogin(username);
+      if (!user) {
+        return cb(null, false, {
+          message: 'Incorrect username or password.',
+        });
+      }
+
+      const valid = validatePassword(password, user.password, user.salt);
+
+      if (!valid) {
+        return cb(null, false, {
+          message: 'Incorrect username or password.',
+        });
+      }
+
+      return cb(null, user);
+    } catch (err) {
+      return cb(err);
+    }
+  })
+);
+
+passport.use(
+  new GitHubStrategy(
+    {
+      clientID: process.env.GITHUB_CLIENT_ID,
+      clientSecret: process.env.GITHUB_CLIENT_SECRET,
+      callbackURL: 'http://localhost:5000/auth/github/callback',
```

Comment:
Consider adding this as a `.env` variable. See previous comments for reasons why this is a good practice.